### PR TITLE
Added sorting to filter endpoint

### LINKED ISSUE #215: Add sorting to entity filter
The second part of https://github.com/nutanix/papiea-js/issues/195 which was not addressed in the PR. Lets add sorting. 

We should support arbitrary nested sorting (meaning sub-sorting), and both ASC/DESC sorting for each.

### PR DESCRIPTION
**Pull request by Emulebest**
resolves #215

### CODE REVIEW

File: papiea-engine/src/entity/entity_api_impl.ts [hunk 1]
```diff
@@ -87,16 +87,16 @@ export class Entity_API_Impl implements Entity_API {
         return [metadata, status];
     }
 
-    async filter_entity_spec(user: UserAuthInfo, kind_name: string, fields: any): Promise<[Metadata, Spec][]> {
+    async filter_entity_spec(user: UserAuthInfo, kind_name: string, fields: any, sortParams: SortParams): Promise<[Metadata, Spec][]> {
         fields.metadata.kind = kind_name;
-        const res = await this.spec_db.list_specs(fields);
+        const res = await this.spec_db.list_specs(fields, sortParams);
         const filteredRes = await this.authorizer.filter(user, res, ReadAction, x => { return { "metadata": x[0] } });
```
Comment (adolgarev) [inline papiea-engine/src/entity/entity_api_impl.ts]: this.authorizer.filter changes the order, please ensure it doesn't

File: papiea-engine/src/databases/spec_db_interface.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { Metadata, Spec, Entity_Reference } from "papiea-core";
+import { SortParams } from "../utils/utils";
```
Comment (adolgarev) [inline papiea-engine/src/databases/spec_db_interface.ts]: utils usually becomes a junkyard. SortParams is a part of Entity_API then it should belong there

File: papiea-engine/src/databases/spec_db_mongo.ts [hunk 3]
```diff
@@ -73,7 +74,7 @@ export class Spec_DB_Mongo implements Spec_DB {
         return [result.metadata, result.spec];
     }
 
-    async list_specs(fields_map: any): Promise<([Metadata, Spec])[]> {
+    async list_specs(fields_map: any, sortParams: SortParams): Promise<([Metadata, Spec])[]> {
```
Comment (shlomi-nx) [inline papiea-engine/src/databases/spec_db_mongo.ts]: The interface treats `sortParams` as optional

File: papiea-engine/src/utils/utils.ts [hunk 4]
```diff
@@ -65,6 +66,31 @@ export function processPaginationParams(offset: number | undefined, limit: numbe
 
 }
 
+export function processSortQuery(query: string | undefined) {
+    if (query === undefined) {
+        return {};
```
Comment (shlomi-nx) [inline papiea-engine/src/utils/utils.ts]: Perhaps this should return `undefined`?

**Pull request merged**